OMEdit/OMEdit/OMEditGUI/Modeling/ModelWidgetContainer.cpp

```diff
@@ -2992,11 +2996,53 @@ void GraphicsView::keyPressEvent(QKeyEvent *event)
     removeCurrentConnection();
   } else if (event->key() == Qt::Key_Escape && isCreatingTransition()) {
     removeCurrentTransition();
+  } else if (event->key() == Qt::Key_Escape && isCreatingShape()) {
+    ShapeAnnotation *shapeToDelete = nullptr;
+    bool *isCreatingShape = nullptr;
+    if ( *(isCreatingShape = &mIsCreatingLineShape) ) {
+      shapeToDelete = mpLineShapeAnnotation;
+    } else if ( *(isCreatingShape = &mIsCreatingPolygonShape) ) {
+      shapeToDelete = mpPolygonShapeAnnotation;
+    } else if ( *(isCreatingShape = &mIsCreatingRectangleShape) ) {
+      shapeToDelete = mpRectangleShapeAnnotation;
+    } else if ( *(isCreatingShape = &mIsCreatingEllipseShape) ) {
+      shapeToDelete = mpEllipseShapeAnnotation;
+    } else if ( *(isCreatingShape = &mIsCreatingTextShape) ) {
+      shapeToDelete = mpTextShapeAnnotation;
+    } else if ( *(isCreatingShape = &mIsCreatingBitmapShape) ) {
+      shapeToDelete = mpBitmapShapeAnnotation;
+    }
+    if (shapeToDelete != nullptr) {
+      *isCreatingShape = false;
+      mpModelWidget->getUndoStack()->undo();
+      setIsCreatingPrologue(false);
+      deleteShapeAndSetItToNullptr(shapeToDelete);
+      uncheckAllShapeDrawingActions();
+    } else /*If they are checked but we have yet to start actually creating something*/{
+      uncheckAllShapeDrawingActions();
+    }
```

Comment:
I guess we have decided not to delete/undo the operation. In general with the undo stack we don't delete items. If you delete it then the program will crash on the redo. My suggestion is to just finish the operation when ESC key is pressed. You can do it by creating double click event for line and polygon. For other shapes just call their create functions e.g., `createRectangleShape()`.

Reply:
Reimplemented it after the Dymola reference you provided